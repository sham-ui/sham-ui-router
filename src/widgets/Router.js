import { DI, options } from 'sham-ui';
import RouterWidget from './Router.sht';

export default class extends RouterWidget {
    constructor() {
        super( ...arguments );
        DI.bind( 'widgets:router', this );
        this._lastRenderedPage = null;
    }

    render() {
        super.render( ...arguments );
        const router = DI.resolve( 'router' );
        this.remove();
        ShamUIView.insert(
            this,
            this.container,
            { ref: null },
            router.activePageWidget,
            router.activePageOptions
        );
    }
}