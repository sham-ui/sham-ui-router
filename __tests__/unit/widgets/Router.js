import Router from '../../../src/widgets/Router';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( Router, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
