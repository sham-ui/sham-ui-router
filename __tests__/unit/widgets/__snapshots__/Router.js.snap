// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {},
  "html": "",
  "nodes": Array [],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
